pkg/utils/common.go

```diff
@@ -36,3 +37,29 @@ func SliceFilter[T any](in []T, f FilterFunc[T]) []T {
 func LatticeServiceName(name string, namespace string) string {
 	return fmt.Sprintf("%s-%s", Truncate(name, 20), Truncate(namespace, 18))
 }
+
+func PointerStringSlicesElementsEqual(arr1, arr2 []*string) bool {
+	if len(arr1) != len(arr2) {
+		return false
+	}
+
+	map1 := make(map[string]int)
+	map2 := make(map[string]int)
+
+	for _, str := range arr1 {
+		if str != nil {
+			map1[*str]++
+		} else {
+			map1["<nil>"]++
+		}
+	}
+
+	for _, str := range arr2 {
+		if str != nil {
+			map2[*str]++
+		} else {
+			map2["<nil>"]++
+		}
+	}
+	return reflect.DeepEqual(map1, map2)
```

Comment:
dont get it, reflect.DeepEqual(arr1, arr2) should return correct result

Reply:
I think this method ignore the ordering of the 2 slices (like a set)

Reply:
It's even more confusing, we dont care about order, but care about duplicates and nil's

Reply:
I dont think it's a generic function, please move it to the place where you do comparison for security groupIds

There is a library for generic slices operations, like equality and sort. https://pkg.go.dev/golang.org/x/exp@v0.0.0-20230905200255-921286631fa9/slices

you can combine with SliceMap from common.go to convert pointers to strings, for example:

```go
func securityGroupIdsEqual(arr1, arr2 []*string) bool {
  ids1 := slices.Sort(SliceMap(arr1, aws.String))
  ids2 := slices.Sort(SliceMap(arr2, aws.String))
  return slices.Equal(ids1, ids2)
}
```

I dont get intention for comparison, how "equal" id's should be. I would guess we compare only non empty strings, and non-nil pointers. Might be worth to filter them out, then sort or put into set. But using set here seems redundant since IDs suppose to be unique, isnt it?

Reply:
Thanks, changed to securityGroupIdsEqual()